### PR TITLE
Feat/dev 1472 upgrade pre commit hooks

### PR DESCRIPTION
**Pull request by qqiao2024**
upgrade pre-commit-hooks in pre-commit-config and reforamt.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -57,6 +57,7 @@ docs/_build/
 #*#
 
 #venv
```
Comment (mpsolano) [inline .gitignore:59]: Should the comment be removed since venv has been added?

**Review by mpsolano — APPROVED** [review]
Looks good!

File: .gitignore (hunk 1 above)
Comment (qqiao2024) [reply .gitignore:59]: I think that comment is just trying to say this is a section for virtualenv.

Comment (mpsolano) [reply .gitignore:59]: Ah, that makes sense. Thank you!

**Review by ProfOak — APPROVED** [review]

**Review by stilesj-uchicago — APPROVED** [review]

**Pull request merged**